addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Cluster"
+// +kubebuilder:printcolumn:name="DISPLAY NAME",type=string,JSONPath=`.spec.addOnMeta.displayName`
+// +kubebuilder:printcolumn:name="CRD NAME",type=string,JSONPath=`.spec.addOnConfiguration.crdName`
+
+// ClusterManagementAddOn represents the registration of an add-on to the cluster manager.
+// This resource allows the user to discover which add-on is available for the cluster manager and
+// also provides metadata information about the add-on.
+// This resource also provides a linkage to ManagedClusterAddOn, the name of the ClusterManagementAddOn
+// resource will be used for the namespace-scoped ManagedClusterAddOn resource.
+// ClusterManagementAddOn is a cluster-scoped resource.
+type ClusterManagementAddOn struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec represents a desired configuration for the agent on the cluster management add-on.
+	// +required
+	Spec ClusterManagementAddOnSpec `json:"spec"`
+
+	// status represents the current status of cluster management add-on.
+	// +optional
+	Status ClusterManagementAddOnStatus `json:"status,omitempty"`
+}
+
+// ClusterManagementAddOnSpec provides information for the add-on.
+type ClusterManagementAddOnSpec struct {
+	// addOnMeta is a reference to the metadata information for the add-on.
+	// +optional
+	AddOnMeta AddOnMeta `json:"addOnMeta"`
```

Comment:
Do we think that this kind of metadata is going to grow in the future? Would it be easier to just list the title/description/etc at the root of the `spec`?